### PR TITLE
[ALLUXIO-2410] Check consistency flag

### PR DESCRIPTION
**Pull request by calvinjia**
Allows the consistency check to propagate more info to the user and be disabled (enabled by default).
https://alluxio.atlassian.net/browse/ALLUXIO-2410

### CODE REVIEW

**Comment by calvinjia** [comment]
alluxio-bot, check this please.

File: tests/src/test/java/alluxio/master/StartupConsistencyCheckTest.java [hunk 1]
```diff
@@ -76,13 +77,14 @@ public void consistent() throws Exception {
...
     waitForStartupConsistencyCheck(master);
-    Assert.assertTrue(master.getStartupConsistencyCheck().isEmpty());
+    Assert.assertTrue(master.getStartupConsistencyCheck().getInconsistentUris().isEmpty());
   }
 
   /**
    * Tests that an inconsistent Alluxio system's startup check correctly detects the inconsistent
    * files.
    */
+  @Test
```
Comment (aaudiber) [inline tests/src/test/java/alluxio/master/StartupConsistencyCheckTest.java]: ^^;

File: docs/_data/table/master-configuration.csv [hunk 2]
```diff
@@ -15,6 +15,7 @@ alluxio.master.lineage.recompute.interval.ms,600000
 alluxio.master.lineage.recompute.log.path,${alluxio.logs.dir}/recompute.log
 alluxio.master.port,19998
 alluxio.master.retry,29
+alluxio.master.startup.consistencycheck.enabled,true
```
Comment (aaudiber) [inline docs/_data/table/master-configuration.csv]: Maybe `consistency.check`? We aren't 100% consistent about this, but most of the time we put periods at word boundaries.

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 3]
```diff
@@ -425,28 +425,74 @@ public void start(boolean isLeader) throws IOException {
...
+   * Class to represent the status and result of the startup consistency check.
    */
-  public List<AlluxioURI> getStartupConsistencyCheck() {
-    if (mStartupConsistencyCheck.isDone()) {
-      try {
-        return mStartupConsistencyCheck.get();
-      } catch (Exception e) {
-        LOG.warn("Failed to complete start up consistency check.", e);
-      }
+  public static class StartupConsistencyCheck {
```
Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: maybe rename to `StartupConsistencyCheckResult`? I was confused for a while thinking that this class actually did the startup consistency check.

File: docs/_data/table/master-configuration.csv (hunk 2 above)
Comment (calvinjia) [inline docs/_data/table/master-configuration.csv]: I took the approach that we would never want to branch off there, but I can go with the majority until we decide to make it consistent across the codebase.

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 3 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: `StartupConsistencyCheckStatus`

File: docs/_data/table/master-configuration.csv (hunk 2 above)
Comment (jsimsa) [inline docs/_data/table/master-configuration.csv]: isn't the convention to use `.` to separate words?

**Comment by jsimsa** [comment]
left some minor comments

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 3 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: but then it stutters with `StartupConsistencyCheckStatus.Status`

File: docs/_data/table/master-configuration.csv (hunk 2 above)
Comment (calvinjia) [inline docs/_data/table/master-configuration.csv]: Is there an established convention we are using? We have things like `tieredstore` and `writetype`.

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 4]
```diff
@@ -479,19 +479,20 @@ public Status getStatus() {
...
    */
-  public StartupConsistencyCheck getStartupConsistencyCheck() {
+  public StartupConsistencyCheckResult getStartupConsistencyCheck() {
     if (!Configuration.getBoolean(PropertyKey.MASTER_STARTUP_CONSISTENCY_CHECK_ENABLED)) {
-      return new StartupConsistencyCheck(StartupConsistencyCheck.Status.DISABLED, null);
+      return new StartupConsistencyCheckResult(StartupConsistencyCheckResult.Status.DISABLED, null);
     }
     if (!mStartupConsistencyCheck.isDone()) {
-      return new StartupConsistencyCheck(StartupConsistencyCheck.Status.RUNNING, null);
+      return new StartupConsistencyCheckResult(StartupConsistencyCheckResult.Status.RUNNING, null);
```
Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: nit: we could make this line happier by having static constructors for each type of result, then we wouldn't have to pass meaningless nulls or write `StartupConsistencyCheckResult` twice

Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: Are you suggesting to have something like `StartupConsistencyCheckResult.FAILURE` as a constant and only have a constructor for the `COMPLETE` case? Another approach I was thinking about was just moving all the logic into the class, but I think its nicer to use it as a data structure.

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 3 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: In that case, I would be in favor of simply `StartupConsistencyCheck`

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 4 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: I'm suggesting methods in `StartupConsistencyCheckResult` like
```java
public static StartupConsistencyCheckResult Failure() {
  return StartupConsistencyCheckResult(Status.FAILURE, null);
}

public static StartupConsistencyCheckResult Success(List<AlluxioURI> inconsistentUris) {
  return new StartupConsistencyCheckResult(Status.SUCCESS, inconsistentUrls);
}
```

and then we can make the constructor private and the call sites become nicer

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by aaudiber** [comment]
just need to fix the checkstyle issues

**Pull request merged**